python/ray/rllib/agents/ppo/ppo.py

```diff
@@ -53,6 +53,10 @@
     # Uses the sync samples optimizer instead of the multi-gpu one. This does
     # not support minibatches.
     "simple_optimizer": False,
+    # Use a centralized value function
+    "use_centralized_vf": False,
+    # how many agents worth of observations to potentially pad the global observation with
+    "max_vf_agents": 100,
```

Comment:
I wonder if you could make this dynamic by using a max-pool op. That might even help performance by reducing the number of weights needed.

Reply:
Hmm, how would that work? Having trouble visualizing this one.

Reply:
Hm I guess you still need some sort of "max" since the tensor size has to be defined ahead of time to be [?, max_agents, obs_size].

From the network perspective doing a softmax would save weights though right? Basically, you could max-pool on the second dimension after applying some hidden layers, so that the input to the central vf is constant size regardless of number of agents.

Reply:
This is a max-pool on the observations though? Seems like that would throw away a lot of valuable information, though I might be missing something here.

Reply:
Oh, I see! I'll give it a go.